cmd/githubrunnerscalesetlistener/main.go

```diff
@@ -56,12 +66,24 @@ func main() {
 	var c github.Config
```

Comment:
NIT: Should we keep the var `c github.Config` around? Feels like this PR migrates most of the config to the `RunnerScaleSetListenerConfig` struct.

Reply:
What I wanted to do here is to define Config that RunnerScaleSetListener needs, but not read it there. Does not matter if they are flags, envs etc, main should know how to read it and we should pass config to the Listener that listener expects. 
Taking similar approach to k8s: https://github.com/kubernetes/kubernetes/blob/3e9e41f82c8757d8851a4f48d8cbe327b641ef41/cmd/kube-apiserver/app/server.go#L367

You define config struct that you need, and leave it to the caller to decide if it will use env, flags, maybe viper etc...